src/lib/plans/other/AgentLoop.js

```diff
@@ -119,10 +120,39 @@ function chooseBestOption() {
 }
 
 export function updateCarriedParcelsScore() {
-    let sumScore = carriedParcels.reduce((previous, current, index) => previous + current.reward, 0)
-    let closestDelivery = findClosestDelivery()
+    let sumScore = carriedParcels.reduce((previous, current, index) => previous + current.reward, 0)   
     if (sumScore > 0){
-        console.log('updateCarriedParcelsScore', sumScore - closestDelivery.distance, sumScore, closestDelivery.distance)
-        agent.changeIntentionScore('go_deliver', [], sumScore - Math.floor(closestDelivery.distance / 2), 'go_deliver')
+        decidePickupOrDeliver(sumScore,carriedParcels);
     }
-}
\ No newline at end of file
+}
+
+function decidePickupOrDeliver(sumScore,carriedParcels){
+    const lowestValuedParcel = carriedParcels.reduce((previous, current) => previous.reward < current.reward ? previous : current, carriedParcels[0]);
+
+    const nextParcel = Array.from(parcels.values()).reduce((previous, current) => previous.reward > current.reward ? previous : current, parcels.values().next().value);
```

Comment:
This can be improved, we should be computing the reward compared to the distance, not just the reward of the parcel itself.
In case there is a parcel with value 30 at a distance of 10, and one with value 29 at distance 1.